android/app/src/main/java/com/sample/component/NativeComponentActivity.kt

```diff
@@ -0,0 +1,73 @@
+package com.sample.component
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import com.facebook.react.ReactNativeHost
+import com.facebook.soloader.SoLoader
+import dagger.android.AndroidInjection
+import javax.inject.Inject
+
+class NativeComponentActivity : AppCompatActivity() {
+
+    @Inject
+    lateinit var reactNativeHost: ReactNativeHost
```

Comment:
Ensure that `reactNativeHost` is receiving correct lifecycle events from the activity.